### DISCUSSION TITLE
Button to copy the un-readable, original URL

### DISCUSSION THREAD

**Discussion by myl7** [spencerwooo/onedrive-vercel-index#483] [Ideas]
Ref: #481

Though readable URL works on most (maybe all) browsers, external softwares may reject them. How about to add a button / enhance the original `Copy Direct URL` button to provide users with a way to get the original finely URL-encoded URL?

One problem of the former plan is that adding a new copying-link button into the button group looks ... duplicated. I would prefer to enhance the original one, for example, add an arrow on its right side which can expand a dropdown, to select the unreadable, original URL. Anyway the design should be checked by the owner @spencerwooo .

**Comment by spencerwooo** [comment]
I would:

- Revert the original `Copy direct link` back to a URL encoded one.
- Add a new tab for the `Customise direct link` popup for formatting the link to a human-readable pretty-print (with some instructions on the side).

Will work on this asap ;)

**Comment by spencerwooo** [comment]
<img width="1608" alt="image" src="https://user-images.githubusercontent.com/32114380/155481595-ea045de4-868e-43e6-8539-44f4922450b0.png">
